src/targets/maven.ts

```diff
@@ -355,14 +391,33 @@ export class MavenTarget extends BaseTarget {
    * @param distDir directory of the distribution.
    * @returns record of required files.
    */
-  private getFilesForMavenPomDist(distDir: string): Record<string, string> {
+  private getFilesForMavenPomDist(distDir: string): Record<string, string | string[]> {
     const moduleName = parse(distDir).base;
-    return {
+    const files = {
       targetFile: join(distDir, this.getTargetFilename(distDir)),
       javadocFile: join(distDir, `${moduleName}-javadoc.jar`),
       sourcesFile: join(distDir, `${moduleName}-sources.jar`),
       pomFile: join(distDir, 'pom-default.xml'),
     };
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}-all.jar`))) {
+      Object.assign(files, { allFile: join(distDir, `${moduleName}-all.jar`)})
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}.module`))) {
+      Object.assign(files, { moduleFile: join(distDir, `${moduleName}.module`) })
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}-metadata.jar`))) {
+      Object.assign(files, { metadataFile: join(distDir, `${moduleName}-metadata.jar`) })
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}.klib`))) {
+      const cinteropFiles: string[] = []
+      readdirSync(distDir).forEach(file => {
```

Comment:
I suggest not to check for the existence of individual files if we are going to read the whole directory just after that. You could do something like the following, using [the `includes` function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes):
```
files = await readFilesAsync()
if (files.includes(fileXXX)) {
  // do sth
}
```

Reply:
You also may want to consider what the output should look like if someone does a kotlin release and a file is missing.